From 32a9753c782daa682563b6f6eb923d3f112d6b94 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Jean-Fran=C3=A7ois=20Fortin=20Tam?= <nekohayo@gmail.com>
Date: Sat, 8 Oct 2022 12:24:02 -0400
Subject: [PATCH] query-editor: Lengthen search activation timeout to improve
 performance

A more realistic default delay avoids spamming the search backends.
While it does not solve every perf issue, it's an improvement.

Fixes issue #1731
---
 src/nautilus-query-editor.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/src/nautilus-query-editor.c b/src/nautilus-query-editor.c
index b671d818f..78719d457 100644
--- a/src/nautilus-query-editor.c
+++ b/src/nautilus-query-editor.c
@@ -84,7 +84,11 @@ static void nautilus_query_editor_changed (NautilusQueryEditor *editor);
 
 G_DEFINE_TYPE (NautilusQueryEditor, nautilus_query_editor, GTK_TYPE_WIDGET);
 
-#define SEARCH_CHANGED_TIMEOUT 150
+/* A hunt-and-peck typist types at 25-35 words per minute, which means 342 to 480ms between strokes.
+ * An average touch typist types at 50-70 wpm, which means 171 to 240ms "under ideal conditions".
+ * A 150ms default search triggering delay is too short even for fast typists in general,
+ * so wait 400ms after typing, to improve performance by not spamming search engines: */
+#define SEARCH_CHANGED_TIMEOUT 400
 
 static void
 update_fts_sensitivity (NautilusQueryEditor *editor)
-- 
GitLab

